jvm/src/test/scala/org/scalacheck/GenSpecification.scala

```diff
@@ -161,6 +161,10 @@ object GenSpecification extends Properties("Gen") {
     l.length == 0
   }
 
+  property("infiniteStream") = forAll(infiniteStream(arbitrary[Int]), arbitrary[Int => Boolean]) { (s, p) =>
+    s exists p
+  }
```

Comment:
Yes, I was probably getting a little cute here. I think the likelihood of the old test hanging is incredibly remote given the way arbitrary functions are generating and that is the only case where it wouldn't work. But at the same time it would be incredibly likely to pass for even finite streams with a length of 100, so it isn't like it was a particularly great test. I like the test you suggested and changed it along those lines.